diagrams_demo.py

```diff
@@ -0,0 +1,217 @@
+    for obj in groupObjects:
+        if obj.entities!=[] and obj.childIDs!=[]:
+            str = "with Cluster(\"{}\"):".format(obj.groupID)
+            lines = [obj.entityStr if l==str else l for l in lines]
+    
+    #write template in file
+    with open(tempPath, 'w+', encoding='utf-8') as f:
+        for l in lines:
+            f.write(l+'\n')
+    
+    #remove redundant spaces and empty lines
+    with open(tempPath, 'r', encoding='utf-8') as f:
+        lines = f.readlines()
+        lines = [l.strip() for l in lines]
+        lines = [l for l in lines if l != '']
+        lines = removeDuplicates(lines)
+
+        #get group rank for Cluster
+        def clusterRank(str):
+            name = str.split('"')[1].split('"')[0]
+            for obj in groupObjects:
+                if obj.groupID == name:
+                    return obj.rank
+
+        #import libraries from entity types
+        import_diagrams = "import diagrams\nfrom diagrams import Cluster, Diagram, Edge\nfrom diagrams.generic.blank import Blank\n"
+        for obj in groupObjects:
+            if obj.libraries != []:
+                library = obj.libraries[0].rsplit('.', 1)[0]
+                import_libray ='from {} import *\n'.format(library)
+                import_diagrams = import_diagrams + import_libray
+
+        exe_temp=import_diagrams + '\nwith Diagram(\"{}\", show=True, outformat="png", direction="TB"):\n'.format(diagramName) 
+        space ='    ' #indent
+        rank = 0
+        for i in range(len(lines)):
+            if 'with' in lines[i]:
+                rank = clusterRank(lines[i])
+                lines[i] = space*rank+lines[i]+'\n' #indent according to ranks
+            else:
+                if 'with' in lines[i-1]:
+                    rank = rank+1
+                lines[i] = space * rank + lines[i]+'\n'
+            exe_temp = exe_temp+lines[i]
+
+    #add relationships
+    for rela in relationships:
+        if isinstance(rela,dict):
+            if rela['direction'] == 'one way':
+                rela_str = space + rela['startEntityID'] +' '+'>>'+' '+'Edge()'+' '+'>>'+' '+rela['endEntityID']+'\n'
+                exe_temp = exe_temp + rela_str
+            elif rela['direction'] == 'two way':
+                rela_str = space + rela['startEntityID'] + ' ' + '>>'+' ' + 'Edge()' + ' ' + '<<' + ' ' + rela['endEntityID'] + '\n'
+                exe_temp = exe_temp + rela_str
+    
+    #write exe template into file for checking
+    f = open(tempPath, 'w+', encoding='utf-8')
+    f.write(exe_temp)
+    f.close()
+
+    #run template file and see results
+    exec(exe_temp)
+
+#specify the json file path here:
+if __name__ == "__main__":
+    #first parameter is the path for json file, second parameter is path for output template file
+    main(r'C:\Users\lixug\Desktop\gosourceDiagrams\executor-diagrams\testCases\generated_graph-2.json', r'C:\Users\lixug\Desktop\gosourceDiagrams\executor-diagrams\testCases\template.py')
```

Comment:
MIght be better to make this a read prompt or to set it to some standard values 
`~/executor-diagrams/input.json` and `~/executor-diagrams/output.py`